wayland-cursor/src/lib.rs

```diff
@@ -160,23 +180,57 @@ impl CursorTheme {
             pool_size: INITIAL_POOL_SIZE,
             cursors: Vec::new(),
             backend: conn.backend().downgrade(),
+            fallback: None,
         })
     }
 
     /// Retrieve a cursor from the theme.
     ///
     /// This method returns [`None`] if this cursor is not provided either by the theme, or by one of its parents.
+    ///
+    /// If a fallback is set, it will use the data from fallback
     pub fn get_cursor(&mut self, name: &str) -> Option<&Cursor> {
         match self.cursors.iter().position(|cursor| cursor.name == name) {
             Some(i) => Some(&self.cursors[i]),
             None => {
-                let cursor = self.load_cursor(name, self.size)?;
+                let cursor = match self.load_cursor(name, self.size) {
+                    None => {
+                        let Some(ref fallback) = self.fallback else {
+                            return None;
+                        };
+                        let data = fallback.0(name, self.size)?;
+                        let images = xparser::parse_xcursor(&data)?;
+                        let conn = Connection::from_backend(self.backend.upgrade()?);
+                        Cursor::new(&conn, name, self, &images, self.size)
+                    }
+                    Some(cursor) => cursor,
+                };
                 self.cursors.push(cursor);
                 self.cursors.iter().last()
             }
         }
     }
 
+    /// Set the callback to load the data, the return value of the fallback function should be the source of
+    /// xcursor
+    /// Same as calling the following:
```

Comment:
```suggestion
    /// Set a callback to load the cursor data, in case the system theme is missing a cursor that you need.
    ///
    /// Your callback will be invoked with he name and size of the requested cursor and should return a byte
    /// array with the contents of an `xcursor` file, or `None` if you don't provide a fallback for this cursor.
    ///
    /// For example, this defines a generic fallback cursor image and uses it for all missing cursors:
```